test/functional/treesitter/parser_spec.lua

```diff
@@ -886,14 +886,20 @@ int x = INT_MAX;
   it("can fold via foldexpr", function()
     insert(test_text)
 
-    local levels = exec_lua([[
+    local function get_fold_levels()
+      return exec_lua([[
+        vim.treesitter.get_parser(0, "c")
+        local res = {}
+        for i = 1, vim.api.nvim_buf_line_count(0) do
+          res[i] = vim.treesitter.foldexpr(i)
+        end
+        return res
+      ]])
+    end
+
+    exec_lua([[
       vim.opt.filetype = 'c'
       vim.treesitter.get_parser(0, "c")
```

Comment:
Should that be here again? And do you need to set the filetype if you specify the parser explicitly?

Reply:
No. And yes, since `get_parser` will try to associate `''` with `'c'` and error. We should probably relax this and just ignore.

Reply:
Ok, I've tweaked `get_parser()` so it always returns the parser if one is available and `lang=nil`.